staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_docker_test.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package create
+
+import (
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestCreateSecretDockerRegistry(t *testing.T) {
+	username, password, email, server := "test-user", "test-password", "test-user@example.org", "https://index.docker.io/v1/"
+	secretData, err := handleDockerCfgJSONContent(username, password, email, server)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	secretDataNoEmail, err := handleDockerCfgJSONContent(username, password, "", server)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	tests := map[string]struct {
```

Comment:
Can we improve those unit tests to contain also some fake fileSources, and test beween fileSources or Username, and the handling of the JSON?

Reply:
OK, I was thinking that kubectl create secret parsed the dockercfg but I was wrong :)

Anyway, I would like to test this case: https://github.com/kubernetes/kubernetes/pull/98614/files#diff-515e663fc601270fd240da46a29b12cb47ec262d0f952269651639b09ce75336R280-R284

Reply:
Yup, I didn't add any test cases because the function that handles fileSources will be called from create_secret.go and create_secret_test.go will handle that. Unless you want to migrate some of the test cases from there?